src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/DynamicMethod.cs

```diff
@@ -458,19 +458,15 @@ internal RuntimeMethodHandle GetMethodDescriptor()
 
             // if we are here we passed all the previous checks. Time to look at the arguments
             bool wrapExceptions = (invokeAttr & BindingFlags.DoNotWrapExceptions) == 0;
-            object retValue;
-            if (actualCount > 0)
```

Comment:
Why the removal of `if actualCount > 0`? (and again in `Invoke()` below)

Reply:
Simplified the code and ended up not having any measurable perf impact due to other optimizations that were made.

Reply:
It also means we don't need to persist otherwise-unused locals like _actualCount_ across the method call, which results in smaller stack-spilling. Since _arguments_ is already persisted across the method call, we can leverage it to perform the "was this empty?" check automatically as part of the for loop condition.

Reply:
So the overhead of calling `CheckArguments()` is a wash compared to tracking `actualCount`?

Reply:
Correct! It became cheap enough that the extra check wasn't worthwhile.